Games/Chess/main.cpp

```diff
@@ -65,6 +67,16 @@ int main(int argc, char** argv)
         return 1;
     }
 
+    if (unveil("/home/anon/", "wcb") < 0) {
+        perror("unveil");
+        return 1;
+    }
```

Comment:
anon's home directory should not be hard coded.

A path is already unveiled for loading the user's config file a few lines above (using `Core::ConfigFile::get_for_app("Chess")`) :

```cpp
    if (unveil(config->file_name().characters(), "crw") < 0) {
        perror("unveil");
        return 1;
    }
```

So I'm guessing that you're unveiling the home directory to permit loading PGN files. In which case you can do something similar with `Core::StandardPaths::home_directory()` :

```suggestion
    if (unveil(Core::StandardPaths::home_directory().characters(), "wcb") < 0) {
        perror("unveil");
        return 1;
    }
```

Reply:
Yes, unveiling so that FilePicker can browse directories, and so PGN files can be written(/read eventually).
I agree, hard coding it was not the way to go.

What I wanted was the ability for FilePicker to have the same browse/read/write rights as the current user. Like what TextEditor does. But looking at TEs code, I didn't find an `unveil()` call and couldn't figure out how to replicate that behavior.
Do you have a solution for how to do this inside of Chess?

Reply:
> Do you have a solution for how to do this inside of Chess?

Does the code I provided work?

Reply:
> Does the code I provided work?

It does, but only for the user's home directory. It's fine like that I guess, but doesn't replicate TextEditor's behavior exactly. Which is maybe good for a chess program.
Either way I'm going with the code you provided for now.

Reply:
> 
> 
> > Does the code I provided work?
> 
> It does, but only for the user's home directory. It's fine like that I guess, but doesn't replicate TextEditor's behavior exactly. Which is maybe good for a chess program.
> Either way I'm going with the code you provided for now.

Ah yeah the TextEditor can browse the entire file system because it doesn't use `unveil`. Presumably the intention was to permit privileged users to use TextEditor for editing system config files which would require access to most of the system.

We have a new File IPC which was implemented along with the browse `b` unveil flag and seems pretty cool, but not something I've had a chance to look into yet.

The solution above is probably easiest and best for now.